integration-tests/deployment/keystone/deploy.go

```diff
@@ -0,0 +1,807 @@
+}
+
+func registerNOPS(ctx context.Context, req registerNOPSRequest) (*registerNOPSResponse, error) {
+	nops := req.nops
+	tx, err := req.registry.AddNodeOperators(req.chain.DeployerKey, nops)
+	if err != nil {
+		err = DecodeErr(kcr.CapabilitiesRegistryABI, err)
+		return nil, fmt.Errorf("failed to call AddNodeOperators: %w", err)
+	}
+	// for some reason that i don't understand, the confirm must be called before the WaitMined or the latter will hang
+	// (at least for a simulated backend chain)
+	_, err = req.chain.Confirm(tx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to confirm AddNodeOperators confirm transaction %s: %w", tx.Hash().String(), err)
+	}
+
+	receipt, err := bind.WaitMined(ctx, req.chain.Client, tx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to mine AddNodeOperators confirm transaction %s: %w", tx.Hash().String(), err)
+	}
+	if len(receipt.Logs) != len(nops) {
+		return nil, fmt.Errorf("expected %d log entries for AddNodeOperators, got %d", len(nops), len(receipt.Logs))
+	}
+	resp := &registerNOPSResponse{
+		nops: make([]*capabilities_registry.CapabilitiesRegistryNodeOperatorAdded, len(receipt.Logs)),
+	}
+	for i, log := range receipt.Logs {
+		o, err := req.registry.ParseNodeOperatorAdded(*log)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse log %d for operator added: %w", i, err)
+		}
+		resp.nops[i] = o
+	}
+
+	return resp, nil
+}
+
+func defaultCapConfig(capType uint8, nNodes int) *capabilitiespb.CapabilityConfig {
+	switch capType {
+	// TODO: use the enum defined in ??
+	case uint8(0): // trigger
+		return &capabilitiespb.CapabilityConfig{
+			DefaultConfig: values.Proto(values.EmptyMap()).GetMapValue(),
+			RemoteConfig: &capabilitiespb.CapabilityConfig_RemoteTriggerConfig{
+				RemoteTriggerConfig: &capabilitiespb.RemoteTriggerConfig{
+					RegistrationRefresh: durationpb.New(20 * time.Second),
+					RegistrationExpiry:  durationpb.New(60 * time.Second),
+					// F + 1; assuming n = 3f+1
+					MinResponsesToAggregate: uint32(nNodes/3) + 1,
+				},
+			},
+		}
+	case uint8(2): // consensus
+		return &capabilitiespb.CapabilityConfig{
+			DefaultConfig: values.Proto(values.EmptyMap()).GetMapValue(),
+		}
+	case uint8(3): // target
+		return &capabilitiespb.CapabilityConfig{
+			DefaultConfig: values.Proto(values.EmptyMap()).GetMapValue(),
+			RemoteConfig: &capabilitiespb.CapabilityConfig_RemoteTargetConfig{
+				RemoteTargetConfig: &capabilitiespb.RemoteTargetConfig{
+					RequestHashExcludedAttributes: []string{"signed_report.Signatures"}, // TODO: const defn in a common place
+				},
+			},
+		}
+	default:
+		return &capabilitiespb.CapabilityConfig{
+			DefaultConfig: values.Proto(values.EmptyMap()).GetMapValue(),
+		}
+	}
+}
+
+// encodedABI is the abi of the contract that emitted the error
+// hexErrBytes is the hex encoded error data
+// returns an error that is the decoded error
+func parseContractErr(encodedABI string, hexErr string) error {
```

Comment:
should be able to use shared helpers here https://github.com/smartcontractkit/chainlink/blob/cd7c32c0b7401858966ac20de4a3fca74673ab71/integration-tests/deployment/helpers.go#L66